src/plugins/vis_builder/public/visualizations/table/to_expression.ts

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SchemaConfig } from '../../../../visualizations/public';
+import { TableVisExpressionFunctionDefinition } from '../../../../vis_type_table_new/public';
+import { AggConfigs, IAggConfig } from '../../../../data/common';
+import { buildExpression, buildExpressionFunction } from '../../../../expressions/public';
+import { RootState } from '../../application/utils/state_management';
+import { TableOptionsDefaults } from './table_viz_type';
+import { getAggExpressionFunctions } from '../common/expression_helpers';
+import { OpenSearchaggsExpressionFunctionDefinition } from '../../../../data/public';
+
+// TODO: Update to the common getShemas from src/plugins/visualizations/public/legacy/build_pipeline.ts
+//  And move to a common location accessible by all the visualizations
+const getVisSchemas = (
```

Comment:
It is different since table has more `Options`.  We could leave this for now. When we have more and more visualization type, we could see how to make this more reusable.

metrics:

```
const getVisSchemas = (aggConfigs: AggConfigs): any => {
  const createSchemaConfig = (accessor: number, agg: IAggConfig): SchemaConfig => {
    const hasSubAgg = [
      'derivative',
      'moving_avg',
      'serial_diff',
      'cumulative_sum',
      'sum_bucket',
      'avg_bucket',
      'min_bucket',
      'max_bucket',
    ].includes(agg.type.name);

    const formatAgg = hasSubAgg
      ? agg.params.customMetric || agg.aggConfigs.getRequestAggById(agg.params.metricAgg)
      : agg;

    const params = {};

    const label = agg.makeLabel && agg.makeLabel();

    return {
      accessor,
      format: formatAgg.toSerializedFieldFormat(),
      params,
      label,
      aggType: agg.type.name,
    };
  };

  let cnt = 0;
  const schemas: any = {
    metric: [],
  };

  if (!aggConfigs) {
    return schemas;
  }

  const responseAggs = aggConfigs.getResponseAggs();
  responseAggs.forEach((agg) => {
    const schemaName = agg.schema;

    if (!schemaName) {
      cnt++;
      return;
    }

    if (!schemas[schemaName]) {
      schemas[schemaName] = [];
    }

    schemas[schemaName]!.push(createSchemaConfig(cnt++, agg));
  });

  return schemas;
};
```
table
```
const getVisSchemas = (
  aggConfigs: AggConfigs,
  showPartialRows: boolean,
  showMetricsAtAllLevels: boolean
): any => {
  const createSchemaConfig = (accessor: number, agg: IAggConfig): SchemaConfig => {
    const hasSubAgg = [
      'derivative',
      'moving_avg',
      'serial_diff',
      'cumulative_sum',
      'sum_bucket',
      'avg_bucket',
      'min_bucket',
      'max_bucket',
    ].includes(agg.type.name);

    const formatAgg = hasSubAgg
      ? agg.params.customMetric || agg.aggConfigs.getRequestAggById(agg.params.metricAgg)
      : agg;

    const params = {};

    const label = agg.makeLabel && agg.makeLabel();

    return {
      accessor,
      format: formatAgg.toSerializedFieldFormat(),
      params,
      label,
      aggType: agg.type.name,
    };
  };

  let cnt = 0;
  const schemas: any = {
    metric: [],
  };

  if (!aggConfigs) {
    return schemas;
  }

  const responseAggs = aggConfigs.getResponseAggs().filter((agg: IAggConfig) => agg.enabled);
  const metrics = responseAggs.filter((agg: IAggConfig) => agg.type.type === 'metrics');
  const isHierarchical = showMetricsAtAllLevels;
  responseAggs.forEach((agg) => {
    let skipMetrics = false;
    const schemaName = agg.schema;

    if (!schemaName) {
      cnt++;
      return;
    }

    if (schemaName === 'split_row' || schemaName === 'split_column') {
      skipMetrics = responseAggs.length - metrics.length > 1;
    }

    if (!schemas[schemaName]) {
      schemas[schemaName] = [];
    }

    if (!isHierarchical || agg.type.type !== 'metrics') {
      schemas[schemaName]!.push(createSchemaConfig(cnt++, agg));
    }
    if (isHierarchical && (agg.type.type !== 'metrics' || metrics.length === responseAggs.length)) {
      metrics.forEach((metric: any) => {
        const schemaConfig = createSchemaConfig(cnt++, metric);
        if (!skipMetrics) {
          schemas.metric.push(schemaConfig);
        }
      });
    }
  });

  return schemas;
};
```